src/mcts/node.h

```diff
@@ -305,6 +326,8 @@ class Node {
   // Best and worst result for this node.
   GameResult lower_bound_ : 2;
   GameResult upper_bound_ : 2;
+  // Whether the child_ is actually an array of equal length to edges.
+  bool solid_children_ : 1;
```

Comment:
Definitely for later, but any thoughts on what else can be done when all children are visited? E.g., `GetFpu` is unnecessary saving a `sqrt` for reduction.

Reply:
FYI - solid_children doesn't mean all children are visited - it means all children are allocated.  They can still have 0 visits.

This change is not memory neutral for small values of threshold - memory usage goes up a lot because there are many unused nodes in each node block.  In practice the ideal value doesn't cause a large memory usage increase though, since performing those memory allocations is itself expensive if you start doing it for every node.
So there is unlikely to be much logic that can depend on this flag.